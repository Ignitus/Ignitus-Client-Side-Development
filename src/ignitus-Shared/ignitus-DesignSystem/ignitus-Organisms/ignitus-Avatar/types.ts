export type avatar = {
  name: string;
  src: string;
};

/** As AvatarImage represents img feel free to add more props here, as <img /> supports lot of other attributes. */
export type AvatarImageProps = {
  src: string;
  alt: string;
};
