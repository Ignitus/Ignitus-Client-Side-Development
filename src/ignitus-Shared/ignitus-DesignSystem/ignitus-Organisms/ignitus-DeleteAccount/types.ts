export interface PropTypes {
  username: string;
}
