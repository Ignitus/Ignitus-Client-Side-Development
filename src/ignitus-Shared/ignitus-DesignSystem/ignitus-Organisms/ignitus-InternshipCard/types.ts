export type CardTypes = {
  ImageLink: string;
  Title: string;
  Position?: string;
  Place?: string;
  Company?: string;
  Description?: string;
  type?: string;
  JobType?: string;
};

export type CardProps = {
  ImageLink: string;
  Title: string;
  Position?: string;
  Place?: string;
  Company?: string;
  Description?: string;
  type: string;
  JobType?: string;
};

export type StyleProps = {
  size?: string;
};
