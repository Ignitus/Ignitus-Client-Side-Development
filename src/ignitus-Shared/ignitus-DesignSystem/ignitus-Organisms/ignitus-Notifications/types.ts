export type InputTypes = {
  textColor: string;
  titleColor: string;
  bgColor: string;
  picture?: string;
  title?: string;
  textLeft?: string;
  textRight?: string;
};

export type StyledTagProps = {
  bgColor: string;
  color: string;
};
