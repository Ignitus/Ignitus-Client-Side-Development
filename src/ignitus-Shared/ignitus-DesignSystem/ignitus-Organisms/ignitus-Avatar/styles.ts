import styled from '@emotion/styled';

export const AvatarImage = styled.img`
  max-width: 9rem;
`;
