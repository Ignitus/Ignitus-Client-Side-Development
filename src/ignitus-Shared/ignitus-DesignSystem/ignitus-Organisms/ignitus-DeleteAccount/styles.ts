import styled from '@emotion/styled';
import * as C from '../../ignitus-Atoms/colors';
import { Paragraph } from '../../ignitus-Atoms/typography';
import * as F from '../../ignitus-Atoms/fonts';
import { flexibleRowDiv } from '../../shared';
import { Icon } from '../../../ignitus-Utilities/Components/icon';
import { DefaultCheckbox } from '../../..';
import { RoundedButton } from '../../ignitus-Atoms/buttons';

export const DeleteContainer = styled.div`
  background-color: ${C.White};
  background-size: contain;
  border-radius: 1rem;
  padding: 2rem;
  display: inline-flex;
  flex-direction: column;
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  width: 40rem;
`;

export const TopSection = styled(flexibleRowDiv)`
  justify-content: space-between;
`;

export const MiddleSection = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
`;

export const BottomSection = styled(flexibleRowDiv)`
  justify-content: flex-end;
  margin-top: 2rem;
`;

export const StyledIcon = styled(Icon)`
  width: 1.4rem;
  fill: ${props => props.color};
  transition: 0.3s;
  flex: 1;
  margin-top: 0.2rem;
`;

export const IconContainer = styled.button`
  border: none;
  cursor: pointer;
  overflow: hidden;
  margin: 0 0.2rem;
  height: 2.8rem;
  width: 2.8rem;
  border-radius: 50%;
  background-color: ${props => props.color};
  outline-color: transparent;
`;

export const StyledParagraph = styled(Paragraph)`
  padding: 1rem 0;
`;

export const StyledCheckbox = styled(DefaultCheckbox)`
  font-size: ${F.SM};
  margin-bottom: 1rem;
`;

export const StyledTextArea = styled.textarea`
  background-color: ${C.GreyLight};
  padding: 1rem;
  color: ${C.GreySecondaryText};
  font-weight: ${F.SemiBold};
  font-size: ${F.MD};
  border: none;
  border-radius: 1rem;

  ::placeholder {
    color: ${C.GreySecondaryText};
    font-weight: ${F.SemiBold};
    font-size: ${F.MD};
    opacity: 0.5;
  }
`;

export const StyledRoundedButton = styled(RoundedButton)`
  font-weight: ${F.Bold};
  box-shadow: 0 6px 12px ${C.boxShadowColorLight};
`;
