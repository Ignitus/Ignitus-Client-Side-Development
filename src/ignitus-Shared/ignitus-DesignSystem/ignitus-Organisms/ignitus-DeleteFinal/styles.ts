import styled from '@emotion/styled';
import * as C from '../../ignitus-Atoms/colors';
import { Paragraph } from '../../ignitus-Atoms/typography';
import * as F from '../../ignitus-Atoms/fonts';
import { flexibleRowDiv } from '../../shared';
import { Icon } from '../../../ignitus-Utilities/Components/icon';
import { RoundedButton } from '../../ignitus-Atoms/buttons';

export const DeleteContainer = styled.div`
  background-color: ${C.White};
  background-size: contain;
  border-radius: 1rem;
  display: inline-flex;
  flex-direction: column;
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  width: 45rem;
  padding: 2rem;
`;

export const TopSection = styled(flexibleRowDiv)`
  justify-content: space-between;
`;

export const MiddleSection = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
`;

export const BottomSection = styled(flexibleRowDiv)`
  margin-top: 1rem;
  justify-content: flex-end;
`;

export const StyledIcon = styled(Icon)`
  width: 1.4rem;
  fill: ${props => props.color};
  transition: 0.3s;
  flex: 1;
  margin-top: 0.2rem;
`;

export const IconContainer = styled.button`
  border: none;
  cursor: pointer;
  overflow: hidden;
  margin: 0 0.2rem;
  height: 2.8rem;
  width: 2.8rem;
  border-radius: 50%;
  background-color: ${props => props.color};
  outline-color: transparent;
`;

export const StyledParagraph = styled(Paragraph)`
  padding: 0.5rem 0;
`;

export const StyledText = styled(Paragraph)`
  font-weight: ${F.Bold};
  padding: 0.5rem 0;
`;

export const StyledRoundedButton = styled(RoundedButton)`
  box-shadow: 0 6px 12px ${C.boxShadowColorLight};
`;
