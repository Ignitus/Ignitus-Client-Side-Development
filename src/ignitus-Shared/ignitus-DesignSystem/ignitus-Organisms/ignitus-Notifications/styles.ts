import styled from '@emotion/styled';
import { Icon as I } from '../../../ignitus-Utilities/Components/icon';
import { StyledTagProps } from './types';
import { Paragraph } from '../../..';
import { Bold } from '../../ignitus-Atoms/fonts';

export const Container = styled.span<StyledTagProps>`
  background: ${props => props.bgColor};
  color: ${props => props.color};
  border-radius: 0.5rem;
  margin: 0.5rem;
  cursor: pointer;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  box-shadow: 0px 3px 6px rgba(0, 0, 0, 0.1);
`;

export const IconContainer = styled.div`
  display: flex;
`;

export const Icon = styled(I)`
  width: 1.5rem;
  fill: ${props => props.color};
  transition: 0.3s;
`;

export const Image = styled.img`
  height: 3rem;
  border-radius: 50%;
  margin-right: 1rem;
`;

export const Text = styled(Paragraph)`
  font-weight: ${Bold};
  color: ${props => props.color};
`;

export const LeftContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

export const RightContainer = styled.div``;
