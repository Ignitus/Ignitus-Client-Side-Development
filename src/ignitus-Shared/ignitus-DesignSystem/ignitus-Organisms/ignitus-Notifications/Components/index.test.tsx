import React from 'react';
import { mount, render } from 'enzyme';
import emotionSerializer, { matchers } from 'jest-emotion';
import { Notifications } from './index';
import * as S from '../styles';
import { InputTypes } from '../types';
import * as C from '../../../ignitus-Atoms/colors';

const props: InputTypes = {
  textColor: `${C.Black}`,
  titleColor: `${C.IgnitusBlue}`,
  bgColor: `${C.White}`,
  picture:
    'https://storage.googleapis.com/ignitus_assets/ig-avatars/eugene.png',
  title: '3 mentors',
  textLeft: 'We found',
  textRight: 'that could help you with your journey',
};

expect.addSnapshotSerializer(emotionSerializer);
expect.extend(matchers);

it('should rendered', () => {
  const tree = render(<Notifications {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<Notifications />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Notifications {...props} />);
  });

  it('should render TextColor, BgColor', () => {
    expect(wrapper.find(S.Container).prop('color')).toBe(props.textColor);
    expect(wrapper.find(S.Container).prop('bgColor')).toBe(props.bgColor);
  });

  it('should render Image, CrossIcon', () => {
    expect(wrapper.find(S.Image).length).toBe(1);
    expect(wrapper.find(S.Image).prop('src')).toBe(props.picture);
    expect(wrapper.find(S.Icon).length).toBe(1);
  });

  it('should render Text with correct color', () => {
    const value = `\u00A0 ${props.title} \u00A0`;
    expect(wrapper.find(S.Text).text()).toBe(value);
    expect(wrapper.find(S.Text).prop('color')).toBe(props.titleColor);
  });

  it('should render TextLeft, TextRight', () => {
    const value = `\u00A0 ${props.title} \u00A0`;
    expect(wrapper.find(S.LeftContainer).text()).toBe(
      props.textLeft + value + props.textRight,
    );
  });
});
