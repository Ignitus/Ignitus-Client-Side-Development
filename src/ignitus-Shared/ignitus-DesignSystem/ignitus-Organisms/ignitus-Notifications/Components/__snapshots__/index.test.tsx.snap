// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered 1`] = `
.emotion-6 {
  background: #ffffff;
  color: #000000;
  border-radius: 0.5rem;
  margin: 0.5rem;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 3px 6px rgba(0,0,0,0.1);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  height: 3rem;
  border-radius: 50%;
  margin-right: 1rem;
}

.emotion-1 {
  font-size: 16px;
  font-weight: 400;
  color: #000066;
  margin: 0;
  font-weight: 700;
  color: #000066;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  width: 1.5rem;
  fill: #9C9C9C;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

<span
  class="emotion-6"
  color="#000000"
>
  <div
    class="emotion-2"
  >
    <img
      class="emotion-0"
      src="https://storage.googleapis.com/ignitus_assets/ig-avatars/eugene.png"
    />
    We found
    <p
      class="emotion-1"
      color="#000066"
    >
        3 mentors  
    </p>
    that could help you with your journey
  </div>
  <div
    class="emotion-5"
  >
    <div
      class="emotion-4"
    >
      <svg
        aria-label="CrossIcon"
        class="emotion-3"
        color="#9C9C9C"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          CrossIcon
        </title>
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M18.3 5.71c-.39-.39-1.02-.39-1.41 0L12 10.59 7.11 5.7c-.39-.39-1.02-.39-1.41 0-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
        />
      </svg>
    </div>
  </div>
</span>
`;
